### ISSUE TITLE
Report "unsafe overlap" of protocol in `isinstance` or `issubclass` usage

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6882]
The typing spec indicates that type checkers should generate an error when `isinstance` or `issubclass` is used with a runtime-checkable protocol and there is a possibility of an "unsafe overlap" between the tested value and the protocol.

```python
# > Type checkers should reject an isinstance() or issubclass() call if there
# > is an unsafe overlap between the type of the first argument and the protocol.

@runtime_checkable
class Proto3(Protocol):
    def method1(self, a: int) -> int:
        ...

class Concrete3A:
    def method1(self, a: str) -> None:
        pass

class Concrete3B:
    method1: int = 1

def func3():
    if isinstance(Concrete3A(), Proto3):  # Type error: unsafe overlap
        pass

    if isinstance(Concrete3B(), Proto3):  # Type error: unsafe overlap
        pass

    if issubclass(Concrete3A, Proto3):  # Type error: unsafe overlap
        pass
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
